dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/Filter.java

```diff
@@ -60,11 +60,19 @@ default Result onResponse(Result appResponse, Invoker<?> invoker, Invocation inv
         return appResponse;
     }
 
+    @Deprecated
     interface Listener {
 
+        void onResponse(Result appResponse, Invoker<?> invoker, Invocation invocation);
+
+        void onError(Throwable t, Invoker<?> invoker, Invocation invocation);
+    }
+    interface Listener2 {

```

Comment:
I think keep two methods in Listener is better:
```
interface Listener {
        @Deprecated
        void onResponse(Result appResponse, Invoker<?> invoker, Invocation invocation);

        default void onMessage(Result appResponse, Invoker<?> invoker, Invocation invocation) {
                onResponse(appResponse, invoker, invocation);
        }

        void onError(Throwable t, Invoker<?> invoker, Invocation invocation);
    }
```
Then call the `onMessage` method in `ProtocolFilterWrapper`, so user can implement eithor `onMessage` or `onResponse` method.